test/fake-timers-test.js

```diff
@@ -320,11 +320,22 @@ describe("FakeTimers", function() {
             assert(FakeTimers.evalCalled);
         });
 
-        it("evals non-function callbacks", function() {
-            this.clock.setTimeout("FakeTimers.evalCalled = true", 10);
-            this.clock.tick(10);
+        it("does not accept non-function callbacks", function() {
+            var notTypeofFunction = "FakeTimers.evalCalled = true";
+            this.clock.setTimeout(notTypeofFunction, 10);
```

Comment:
This is where the exception should be thrown, as this is the place where the programmer made the mistake.

Reply:
The use of `eval()` was present in the `callTimer()` function and not in the `addTimer()` function. You are basically right. I'm afraid that changing the `addTimer()` would cause many tests to fails and it might be a radical change. But I'll look into it :)

Reply:
Don't worry. It shouldn't break any existing tests, except the ones that should, because a string is passed to `setTimeout`.

Reply:
There is a function called `addTimer()` that actually creates all these timers:

Here it is:
https://github.com/sinonjs/fake-timers/blob/89e9e9b58c12fdf3301067dee6282ca829c48c56/src/fake-timers-src.js#L265

If I add this code to it:
```
if (typeof timer.func !== "function") {
            throw new TypeError(
                "[ERR_INVALID_CALLBACK]: Callback must be a function."
            );
}
```
It will break these 4 tests: 

https://github.com/sinonjs/fake-timers/blob/89e9e9b58c12fdf3301067dee6282ca829c48c56/test/fake-timers-test.js#L275

https://github.com/sinonjs/fake-timers/blob/89e9e9b58c12fdf3301067dee6282ca829c48c56/test/fake-timers-test.js#L285

https://github.com/sinonjs/fake-timers/blob/89e9e9b58c12fdf3301067dee6282ca829c48c56/test/fake-timers-test.js#L3097

https://github.com/sinonjs/fake-timers/blob/89e9e9b58c12fdf3301067dee6282ca829c48c56/test/fake-timers-test.js#L3087

Reply:
That is correct. These tests pass in a string instead of a function. However, they actually test something else. I guess it's reasonable to change those tests and pass in functions.

Reply:
@benjamingr What do you think?

Reply:
@itayperry I'm not sure what I'm being asked here but everything Maximilian has been saying sounded like useful and reasonable feedback and I encourage you to read these tests by stepping through them with a debugger.